src/test/java/com/ericsson/ei/jmespath/test/TestJmesPathInterface.java

```diff
@@ -31,70 +31,106 @@
+    private final String inputDiffpath = "src/test/resources/DiffFunctionInput.json";
+    private final String extractionRuleFilePath = "src/test/resources/ExtractionRule.txt";
+
+    private final String aggregatedObjectFilePath = "src/test/resources/AggregatedDocument.json";
+
+    static Logger log = (Logger) LoggerFactory.getLogger(TestJmesPathInterface.class);
+
+    @Test
+    public void testRunRuleOnEvent() {
+        unitUnderTest = new JmesPathInterface();
+        String jsonInput = null;
+        String jsonOutput = null;
+        String extractionRulesTest = null;
+        JsonNode output = null;
+
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputFilePath));
+            jsonOutput = FileUtils.readFileToString(new File(outputFilePath));
+            extractionRulesTest = FileUtils.readFileToString(new File(extractionRuleFilePath));
+            ObjectMapper objectmapper = new ObjectMapper();
+            output = objectmapper.readTree(jsonOutput);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+
+        JsonNode result = unitUnderTest.runRuleOnEvent(extractionRulesTest, jsonInput);
+        assertEquals(result, output);
+    }
+
+    @Test
+    public void testDiffFunction() {
+        unitUnderTest = new JmesPathInterface();
+        String jsonInput = null;
+        JsonNode expectedResult = null;
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputDiffpath));
+            ObjectMapper mapper = new ObjectMapper();
+            expectedResult = mapper.readTree("{\"testCaseExecutions\":[{\"testCaseDuration\":6.67}]}");
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        String processRule = "{testCaseExecutions :[{testCaseDuration : diff(testCaseExecutions[0].testCaseFinishedTime, testCaseExecutions[0].testCaseStartedTime)}]}";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(result, expectedResult);
+    }
+
+    @Test
+    public void testLiteral() {
+        unitUnderTest = new JmesPathInterface();
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode literalJson;
+        try {
+            literalJson = mapper.readTree("{}");
+            JsonNode input = mapper.readTree("{\"id\":\"test\"}");
+            ((ObjectNode) literalJson).put("eventId", "b6ef1hd-25fh-4dh7-b9vd-87688e65de47");
+            String ruleString = literalJson.toString();
+            ruleString = "`" + ruleString + "`";
+            unitUnderTest.runRuleOnEvent(ruleString, input.toString());
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        //        String literal = "{\"eventId\":"`"fb6ef1hd-25fh-4dh7-b9vd-87688e65de47"`"}";
+    }
+
+    @Test
+    public void testAggregatedObject() {
```

Comment:
This tests the functionality of JMESpath as it is it does not test your changes.

Reply:
Done